gnocchi/tests/functional/gabbits/search.yaml

```diff
@@ -21,12 +21,39 @@ tests:
       GET: /v1/search/resource/foobar
       status: 404
 
+    # FIXME(sileht): this test looks wrong, it talks about invalidity
+    # but asserts it return 200...
     - name: search with invalid uuid
       POST: /v1/search/resource/generic
       data:
         =:
           id: "cd9eef"
 
+    - name: search invalid and value
+      desc: and should be followed by a list, not dict
+      POST: /v1/search/resource/generic
+      data:
+        and:
+          project_id: foobar
+      status: 400
+
+    - name: search invalid ne value
+      desc: attribute value for binary operator must not be dict or list
+      POST: /v1/search/resource/generic
+      data:
+        ne:
+          project_id:
+            - foobar
+      status: 400
+
+    - name: search invalid nor value
+      desc: uninary operator must follow by dict, not list
+      POST: /v1/search/resource/generic
+      data:
+        nor:
```

Comment:
Fixed, I have also added some expected response string, to ensure I test the expected query.